### PR TITLE
App module api documentation for list and show[MAD-8830]

### PR DESCRIPTION
**Pull request by Neeraja82**
App module api documentation for list and show[MAD-8830]

[MAD-8830]: https://liquidweb.atlassian.net/browse/MAD-8830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/list.md [hunk 1]
```diff
@@ -1,113 +1,66 @@
...
-**Access**: order view permission, or service edit permission
-
-**Parameters**:
-- string `filter[name]` (optional): filters results by app name.
-- string `filter[version]` (optional): filters results by app version.
+#### Input
+- integer `filter[id]` (optional): filter list by system ID
+- string `filter[name]` (optional): filter list by app name.
+- string `filter[version]` (optional): filter list by app version.
+- integer `match` (optional): match against id value
```
Comment (adrian-enspired) [inline app/list.md:13]: should be match[name] and match[version]

File: app/list.md [hunk 2]
```diff
@@ -1,113 +1,66 @@
...
+- integer `range` (optional): find id values within .. range
```
Comment (adrian-enspired) [inline app/list.md:14]: should be range[type], [id], [name], and [version]

File: app/list.md [hunk 3]
```diff
@@ -1,113 +1,66 @@
...
 - integer `page` (optional): 1-based result set count for paginated results.
 - integer `pageSize` (optional): maximum number of results to include per "page" of a paginated list.
 - string `sortBy` (optional): field to sort the list by; one of `type`|`id`|`name`|`version`.
 - string `sortOrder` (optional): sort direction; one of `ASC`|`DESC`.
 
-**Request**:
+#### Request
 ```
-curl -i "$PORTAL_API_URL/v1/app" \
+$ curl -i -X GET "$PORTAL_API_URL/v1/app" \
```
Comment (adrian-enspired) [inline app/list.md:22]: `-X GET` is unnecessary

File: app/list.md [hunk 4]
```diff
@@ -1,113 +1,66 @@
...
   -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
```
Comment (adrian-enspired) [inline app/list.md:24]: there's no request body in this example, so content-type is not needed

File: app/list.md [hunk 5]
```diff
@@ -1,113 +1,66 @@
...
-Server: nginx
-Date: Wed, 17 Jul 2019 02:59:06 GMT
+Date: Thu, 28 Oct 2021 12:51:27 GMT
 Content-Type: application/json;charset=utf-8
-Content-Length: 705
-Vary: Accept-Encoding
-X-Powered-By: PHP/7.2.15
-Content-Range: items 1-10/10
+Content-Length: 44
+Location: /v1/app
```
Comment (adrian-enspired) [inline app/list.md:35]: there shouldn't be a `Location` header in this response (did you see one?)

File: app/list.md [hunk 6]
```diff
@@ -1,113 +1,66 @@
...
+  . . .
 ]
 ```
 
-**Failure Response** (not logged in): 401 Unauthorized
+**Failure Response** (not logged in, expired token, etc.): 401 Unauthorized
 
 **Failure Response** (insufficient permissions): 403 Forbidden
 
-**Failure Response** (bad range/page number/page size): 416 Requested Range Not Satisfiable
```
Comment (adrian-enspired) [inline app/list.md:111]: should not be removed

please review other "list" endpoints already merged - they can all return `416`

File: app/show.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+#### Access
+app view
+
+#### Request
+```
+$ curl -i -X GET "$PORTAL_API_URL/v1/app/{id}" \
+  -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
+  -H "Accept: application/json" \
```
Comment (adrian-enspired) [inline app/show.md:14]: no trailing `\`

File: app/show.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+Shows App details.
+
+#### Access
+app view
+
+#### Request
+```
+$ curl -i -X GET "$PORTAL_API_URL/v1/app/{id}" \
+  -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
```
Comment (adrian-enspired) [inline app/show.md:13]: no request body, so no `content-type`

**Review by adrian-enspired — CHANGES_REQUESTED** [review]

File: app/show.md (hunk 7 above)
Comment (Neeraja82) [reply app/show.md]: Removed, please review.

File: app/show.md (hunk 8 above)
Comment (Neeraja82) [reply app/show.md:13]: Removed, please review.

**Review by adrian-enspired — APPROVED** [review]

**Pull request merged**